### PR TITLE
Turn tinode.js to nodeJS compliant code

### PR DESCRIPTION
**Pull request by fstn**

### CODE REVIEW

**Comment by fstn** [comment]
I'm not able to format it back to you format... which software do you use to format your code?

**Comment by or-else** [comment]
It's very hard to read the diff because of tabs vs spaces. 
I'm using Atom https://atom.io/. Indent = 2 spaces. In your case indent is 4 spaces. Let's use the same 2 spaces, otherwise it's a hassle. Thanks!

**Comment by fstn** [comment]
I'm sorry I'm really not able to format it like you :(  I downloaded Atom but it does nothing, I tried to reduce to 2 but it's still not the same format

**Comment by fstn** [comment]
It might be a good solution to put a formatter as dev dependency https://www.npmjs.com/package/js-beautify

**Comment by or-else** [comment]
Overall I'm fine with this PR except for the `function()` -> `function ()`

**Comment by or-else** [comment]
If you could add dev dependency on https://www.npmjs.com/package/js-beautify to package.json together with correct settings, that would be great. 

Thanks!

File: src/tinode.js [hunk 1]
```diff
@@ -1369,32 +1444,35 @@ Tinode.isNewGroupTopicName = function(name) {
  *
  * @returns {string} semantic version of the library, e.g. '0.15.5-rc1'.
  */
-Tinode.getVersion = function() {
+Tinode.getVersion = function () {
   return VERSION;
 };
 
+Tinode.setWebSocketProvider = function (provider) {
```
Comment (or-else) [inline src/tinode.js]: Add js doc comment here?

File: src/tinode.js [hunk 2]
```diff
@@ -597,7 +661,7 @@ var Connection = function(host_, apiKey_, transport_, secure_, autoreconnect_) {
 
         log("Connecting to: " + url);
 
-        var conn = new WebSocket(url);
+        var conn = new WebSocketProvider(url);
```
Comment (or-else) [inline src/tinode.js]: Where is WebSocketProvider defined?

File: src/tinode.js [hunk 3]
```diff
@@ -1373,6 +1447,9 @@ Tinode.getVersion = function() {
   return VERSION;
 };
 
+Tinode.setWebSocketProvider = function(provider) {
```
Comment (or-else) [inline src/tinode.js]: Add jsdoc comment here?

File: src/tinode.js (hunk 2 above)
Comment (fstn) [reply src/tinode.js]: At the top

File: src/tinode.js (hunk 1 above)
Comment (fstn) [reply src/tinode.js]: thanks

File: src/tinode.js (hunk 2 above)
Comment (or-else) [reply src/tinode.js]: There got to be a dependency on WebSocketProvider somewhere?

Comment (or-else) [reply src/tinode.js]: I see `let WebSocketProvider` but that's just a variable declaration. There got be something like `require(...)` somewhere?

Comment (fstn) [reply src/tinode.js]: We might add in the doc ... for Node JS call Tinode.setWebSocketProvider(require('ws'));

**Pull request merged**